src/install/windows-shim/bun_shim_impl.zig

```diff
@@ -675,51 +749,114 @@ fn launcher(bun_ctx: anytype) noreturn {
+
+                            break :iteration; // loop back
+                        }
+
+                        if (flags.is_node_or_bun) {
+                            // This script calls for 'bun', but it was not found.
+                            if (dbg)
+                                assert(std.mem.startsWith(u16, std.mem.span(spawn_command_line), comptime wliteral("bun ")));
+                            return mode.fail(.InterpreterNotFoundBun);
+                        }
+                    }
+
+                    // if attempt_number == 1, we already tried rewriting this to bun, and will now fail for real
+                    if (attempt_number == 1) {
+                        if (dbg)
+                            assert(std.mem.startsWith(u16, std.mem.span(spawn_command_line), comptime wliteral("bun ")));
+                        return mode.fail(.InterpreterNotFoundBun);
+                    }
+
+                    // This UTF16 -> UTF-8 conversion is intentionally very lossy, and assuming that ascii text is provided.
+                    // This trade off is made to reduce the binary size of the shim.
+                    failure_reason_argument = brk: {
+                        var i: u32 = 0;
+                        while (spawn_command_line[i] != ' ' and i < 512) : (i += 1) {
+                            failure_reason_data[i] = @as(u7, @truncate(spawn_command_line[i]));
+                        }
+                        break :brk failure_reason_data[0..i];
+                    };
+                    return mode.fail(.InterpreterNotFound);
+                } else return mode.fail(.BinNotFound),
+
+                // TODO: ERROR_ELEVATION_REQUIRED must take a fallback path, this path is potentially slower:
+                // This likely will not be an issue anyone runs into for a while, because it implies
+                // the shebang depends on something that requires UAC, which .... why?
+                //
+                // https://learn.microsoft.com/en-us/windows/security/application-security/application-control/user-account-control/how-it-works#user
+                // https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecutew
+                .ELEVATION_REQUIRED => return mode.fail(.ElevationRequired),
 
-            else => fail(.CreateProcessFailed),
+                else => return mode.fail(.CreateProcessFailed),
+            };
         }
-        comptime unreachable;
-    }
 
-    _ = k32.WaitForSingleObject(process.hProcess, w.INFINITE);
+        _ = k32.WaitForSingleObject(process.hProcess, w.INFINITE);
 
-    var exit_code: w.DWORD = 255;
-    _ = k32.GetExitCodeProcess(process.hProcess, &exit_code);
+        var exit_code: w.DWORD = 255;
+        _ = k32.GetExitCodeProcess(process.hProcess, &exit_code);
 
-    _ = nt.NtClose(process.hProcess);
-    _ = nt.NtClose(process.hThread);
+        _ = nt.NtClose(process.hProcess);
+        _ = nt.NtClose(process.hThread);
```

Comment:
@paperdave was it intentional we close the thread handle only after the process has exited instead of immediately? since this handle is not used, couldn't we close it immediately?